src/components/CodeInput/inputItem.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { StyledInput } from './styled';
+
+const InputItem = React.forwardRef((props, ref) => {
+    const {
+        value,
+        index,
+        disabled,
+        required,
+        readOnly,
+        error,
+        tabIndex,
+        onClick,
+        onChange,
+        onFocus,
+        onBlur,
+        onKeyDown,
+        onPaste,
+        id,
+        isActive,
+    } = props;
+
+    const handleOnChange = (e, inputIndex) => {
+        onChange(e.target.value, inputIndex);
+    };
+
+    const handleOnKeyDown = (e, inputIndex) => {
+        const shouldMovePrev = !value && e.key === 'Backspace';
+        if (shouldMovePrev) {
+            onChange('', inputIndex - 1);
+        }
+        onKeyDown(e);
+    };
+
+    return (
+        <StyledInput
+            key={index}
+            value={value}
+            disabled={disabled}
+            required={required}
+            readOnly={readOnly}
+            error={error}
+            tabIndex={tabIndex}
+            onClick={e => onClick(e, index)}
+            onChange={e => handleOnChange(e, index)}
+            onFocus={e => onFocus(e, index)}
+            onBlur={e => onBlur(e, index)}
```

Comment:
we are sending event and index in the onBlur prop?
if so then fix this since onBlur should only send the native event

Reply:
its ok, I added the index as an extra for the developer if they need the index of the input, might not be necessary at all for them, I'll removed those values.

Reply:
some of those onEvent are needed at the top of the component, inside we are calling the native event only, but we need in special cases the index...

onClick and onBlur are the only ones that can't be called directly

`
onClick={onClick}
onChange={e => handleOnChange(e.target.value, index)} 
onFocus={e => onFocus(e, index)}
onBlur={onBlur}
onKeyDown={e => handleOnKeyDown(e, index)}
`

I am going to change for now the onClick and onBlur, if we need to change the other ones differently please let me know.